airflow/utils/cli.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+Utilities module for cli
+"""
+from __future__ import absolute_import
+from argparse import Namespace
+
+import logging
+import functools
+import os
+import sys
+from datetime import datetime
+
+import airflow.models
+from airflow.utils import action_loggers
+
+
+def action_logging(f):
+    """
+    Decorates function to execute function at the same time submitting action_logging
+    but in CLI context. Also, builds metrics dict and submits action_loggers.
+    In metrics dict:
+        sub_command : name of sub-command
+        start_datetime : start datetime instance by utc
+        end_datetime : end datetime instance by utc
+        full_command : full command line arguments
+        user : current user
+        log : airflow.models.Log DAO instance
+        dag_id : dag id (optional)
+        task_id : task_id (optional)
+        execution_date : execution date (optional)
+        error : exception instance if there's an exception
+
+    :param f: function instance
+    :return: wrapped function
+    """
+    @functools.wraps(f)
+    def wrapper(*args, **kwargs):
+        assert args
+        assert isinstance(args[0], Namespace), \
+            "1st positional argument should be argparse.Namespace instance, " \
+            "but {}".format(args[0])
+        metrics = _build_metrics(f.__name__, args[0])
+
+        try:
+            return f(*args, **kwargs)
+        except Exception:
+            metrics['error'] = sys.exc_info()[1]
+            raise
```

Comment:
Not sure if this should be silenced, but if not then you need to `raise e` as `raise` isn't valid (and will raise the fact that just raise ins't valid I think). It'll look like this: `TypeError: exceptions must be old-style classes or derived from BaseException, not NoneType`

Reply:
The idea I'd like to achieve here is to have decorator function transparent. To achieve this, I think the exception from  actual function needs to be propagated to the caller where an exception from the decorator function needs to be logged but silenced.

I've referenced python reference doc to use "raise" only to re-raise, but I like it to be explicit as you recommended (https://docs.python.org/3/tutorial/errors.html#handling-exceptions). Also, let me add unittest to confirm if it works as intended.